lib/precompiled/01-ecrecover.js

```diff
@@ -27,7 +27,7 @@ module.exports = function (opts) {
 
   var publicKey
   try {
-    publicKey = utils.ecrecover(msgHash, v, r, s)
+    publicKey = utils.ecrecover(msgHash, utils.bufferToInt(v), r, s)
```

Comment:
Thanks @axic.

> This is silently truncating the input to 53 bits.

Afaik the validation happens in `utils.bufferToInt`. It throws for numbers larger than 53 bits, as [it calls `BN().toNumber`](https://github.com/ethereumjs/ethereumjs-util/blob/master/index.js#L184) and this call [throws](https://github.com/indutny/bn.js/blob/master/lib/bn.js#L514) if that's the case. 

> Since v must be either 0 or 1, it should be validated prior 

Should I add some extra validation to make sure it's 0 or 1?

Reply:
> Should some extra validation be added to make sure it's 0 or 1? In case this validation fails, should the function just throw?

There's no other way to fail there unfortunately.

Reply:
> Since v must be either 0 or 1

Sorry, this actually wants 27 or 28 and `ecrecover()` itself will verify the range. I guess `new BN(v).toNumber()` could be a good compromise if we rely on ethereumjs-util to do the validation.

Reply:
cool, I hadn't seen `ecrecover` validated the input too.

I understand then that no further validation is needed here, as `BN` checks for the bit length and `ethereumjs-util` for the actual value?

Reply:
Please use `BN(v).toNumber()` here and not `utils.bufferToInt(v)`